cms/models/managers.py

```diff
@@ -68,10 +68,14 @@ def search(self, q, language=None, current_site_only=True):
         plugins = plugin_pool.get_all_plugins()
         for plugin in plugins:
             cmsplugin = plugin.model
-            if hasattr(cmsplugin, 'search_fields'):
+            if (hasattr(cmsplugin, 'search_fields') and
+                    cmsplugin.cmsplugin_ptr.field.rel.related_name and
+                    cmsplugin.cmsplugin_ptr.field.rel.related_name != '+'):
                 for field in cmsplugin.search_fields:
-                    qp |= Q(**{'placeholders__cmsplugin__%s__%s__icontains' % \
-                               (cmsplugin.__name__.lower(), field): q})
+                    qp |= Q(**{
+                        'placeholders__cmsplugin__{0}_{1}__{2}__icontains'.format(
+                            cmsplugin._meta.app_label, cmsplugin._meta.model_name, field
+                        ): q})
```

Comment:
here we could dynamically use ``cmsplugin.cmsplugin_ptr.field.rel.related_name``, so it does not break if we change the recipe to build related name later (or the user overrides it).